Jenkinsfile

```diff
@@ -68,70 +68,105 @@ pipeline {
-        unstash 'dist'
+    // stage('Integration Test') {
+    //   steps {
+    //     // Run a simple webserver serving the dist folder statically
+    //     // before we run the cypress tests
+    //     writeFile file: 'integration-tests.sh', text: [
+    //       'export PATH=`pwd`/node_modules/.bin:$PATH',
+    //       'http-server -p 4200 dist&',
+    //       'SERVER_PID=$!',
+    //       './scripts/ci/run-integration-tests',
+    //       'RET=$?',
+    //       'echo "cypress exit status: ${RET}"',
+    //       'sleep 10',
+    //       'echo "kill server"',
+    //       'kill $SERVER_PID',
+    //       'exit $RET'
+    //     ].join('\n')
+
+    //     unstash 'dist'
+
+    //     ansiColor('xterm') {
+    //       retry(2) {
+    //         sh '''bash integration-tests.sh'''
+    //       }
+    //     }
+    //   }
+
+    //   post {
+    //     always {
+    //       archiveArtifacts 'cypress/**/*'
+    //       junit 'cypress/results.xml'
+    //     }
+    //   }
+    // }
+
+    // stage('System Test') {
+    //  steps {
+    //    withCredentials([
+    //       [
+    //         $class: 'AmazonWebServicesCredentialsBinding',
+    //         credentialsId: 'f40eebe0-f9aa-4336-b460-b2c4d7876fde',
+    //         accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+    //         secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+    //       ]
+    //     ]) {
+    //      unstash 'dist'
+
+    //      ansiColor('xterm') {
+    //        retry(2) {
+    //          sh '''dcos-system-test-driver -j1 -v ./system-tests/driver-config/jenkins.sh'''
+    //        }
+    //      }
+    //    }
+    //  }
+
+    //  post {
+    //    always {
+    //      archiveArtifacts 'results/**/*'
+    //      junit 'results/results.xml'
+    //    }
+    //  }
+    // }
+
+    // when env.BRANCH_NAME == 'master', we will update the mesosphere:dcos-ui/latest branch
+    // when env.BRANCH_NAME =~ /(tag-name-regex)/, we will open a release PR against dcos/dcos
+    stage('Upload Build') {
+      when {
+        expression { env.BRANCH_NAME == 'master' || env.BRANCH_NAME =~ /^v?(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*)?(\+[0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*)?/ || env.BRANCH_NAME == 'bill/feat/dcos-ui-cd' || env.BRANCH_NAME == 'test-dcos-ui-cd2' }
+      }
 
-        ansiColor('xterm') {
-          retry(2) {
-            sh '''bash integration-tests.sh'''
-          }
+      steps {
+        withCredentials([
+            string(credentialsId: '3f0dbb48-de33-431f-b91c-2366d2f0e1cf',variable: 'AWS_ACCESS_KEY_ID'),
+            string(credentialsId: 'f585ec9a-3c38-4f67-8bdb-79e5d4761937',variable: 'AWS_SECRET_ACCESS_KEY')
+        ]) {
+          sh "BRANCH_NAME=${env.BRANCH_NAME} ./scripts/ci/upload-build"
         }
       }
 
       post {
         always {
-          archiveArtifacts 'cypress/**/*'
-          junit 'cypress/results.xml'
+          archiveArtifacts 'buildinfo.json'
```

Comment:
Do we still need the artifact if everything is done automatically? We don't need to remove it I just wondered